### ISSUE TITLE
cmd/go: go mod vendor and cgo preambles

### ISSUE THREAD

**Issue by wondratsch** [golang/go#29479]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.11.4 openbsd/amd64
go version go1.11.2 linux/amd64
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/wondratsch/.cache/go-build"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/wondratsch/golang"
GOPROXY=""
GORACE=""
GOROOT="/usr/lib/golang"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/golang/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/wondratsch/Downloads/hugo-0.53/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build939105182=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

wget https://github.com/gohugoio/hugo/archive/v0.53.tar.gz
tar -xzf v0.53.tar.gz
cd hugo-0.53
go mod vendor
CGO_ENABLED=1 go build -mod=vendor -a -v -tags extended

### What did you expect to see?

A successful build

### What did you see instead?
# github.com/wellington/go-libsass/libs
vendor/github.com/wellington/go-libsass/libs/encoding.go:4:11: fatal error: sass/context.h: No such file or directory
 // #include "sass/context.h"
           ^~~~~~~~~~~~~~~~
compilation terminated.

### What do you think happened?

cgo can't find it's preamble, I think "go mod vendor" doesn't look for cgo dependencies of build tags. If I run "go build" without the vendor only flag, it works fine.

**Comment by bcmills** [comment]
Dup of #26366.

TL;DR: The `go` tool only considers source files in the same directory as a Go package, for both vendoring and cache invalidation.
